src/Microsoft.AspNetCore.Server.IIS/HttpProtocol.cs

```diff
@@ -63,37 +58,51 @@ public HttpProtocol(PipeFactory pipeFactory, IntPtr pHttpContext)
 
             unsafe
             {
-                var pHttpRequest = NativeMethods.http_get_raw_request(pHttpContext);
+                Method = GetVerb();
+
+                var version = GetVersion();
+
+                RawTarget = GetRawUrl();
+
+                HttpVersion = version.ToString();
```

Comment:
We'll do a performance pass and remove all of these allocations.

Reply:
ToString is wrong for HttpVersion and SslStatus.
https://github.com/aspnet/HttpSysServer/blob/dev/src/Microsoft.AspNetCore.Server.HttpSys/FeatureContext.cs#L170-L183
https://github.com/aspnet/HttpSysServer/blob/dev/src/Microsoft.AspNetCore.Server.HttpSys/RequestProcessing/Request.cs#L228